doc/SAI-Proposal-ICMP-ECHO.md

```diff
@@ -0,0 +1,424 @@
+
+#### sai_clear_icmp_echo_session_stats_fn ####
+This defines the interface to clear session stats.
+It takes the unique identifier icmp_echo_session_id to specify the ICMP ECHO session for which statistics counters are to be cleared. The number_of_counters parameter indicates the count of counters specified in the counter_ids array. The counter_ids array specifies the particular statistics counters to clear.
+
+If the operation is successful, the function returns SAI_STATUS_SUCCESS. Otherwise, it returns a failure status code, indicating the nature of the error.
+
+
+#### sai_icmp_echo_session_state_change_notification_fn #### 
+This is a callback function that is invoked to notify about changes in the state of ICMP ECHO sessions. This callback function is passed as a parameter during the initialization of the switch through the sai_initialize_switch() function.
+
+The callback function takes two parameters:
+
+- count: An integer representing the number of ICMP ECHO session state change notifications in the array.
+- data: An array of sai_icmp_echo_session_state_notification_t structures, each containing information about the state change of an ICMP ECHO session.
+
+#### sai_icmp_echo_api_t #### 
+This defines the ICMP_ECHO API table
+```C
+/** 
+ * @brief ICMP ECHO method table retrieved with sai_api_query() 
+ */ 
+typedef struct _sai_icmp_echo_api_t 
+{ 
+    sai_create_icmp_echo_session_fn            create_icmp_echo_session; 
+    sai_remove_icmp_echo_session_fn            remove_icmp_echo_session; 
+    sai_set_icmp_echo_session_attribute_fn     set_icmp_echo_session_attribute; 
+    sai_get_icmp_echo_session_attribute_fn     get_icmp_echo_session_attribute; 
+    sai_get_icmp_echo_session_stats_fn         get_icmp_echo_session_stats; 
+    sai_clear_icmp_echo_session_stats_fn       clear_icmp_echo_session_stats; 
+
+} sai_icmp_echo_api_t; 
+```
+
+### ICMP echo session with LAG ###
+
+- Sessions across a LAG bundle: Individual sessions are created on each of the member links. The device creates ICMP echo requests for each configured probing session, which are sent through the port specified by the Session attribute SAI_ICMP_ECHO_SESSION_ATT_PORT.
+
+- Session on a logical bundle (LAG interface): A single session is set up on the LAG interface. Given that a LAG interface may have multiple next hops or that a next hop can vary dynamically, the destination for the outbound ICMP echo packets is determined by the router's lookup procedure. The session should be configured with the attributes SAI_ICMP_ECHO_SESSION_ATTR_VIRTUAL_ROUTER and SAI_ICMP_ECHO_SESSION_ATTR_HW_LOOKUP_VALID both set to true.
+
+- Incoming ICMP echo response packets can arrive on any of the member links. If necessary, the Session attribute SAI_ICMP_ECHO_SESSION_ATTR_RX_PORT is provided to verify the incoming packets on a specific member link.
+
+### 4.0 Examples ###
+
+#### 4.0.1 Create session using egress interface: ####
```

Comment:
Please provide examples with LAG

Reply:
Added a section "ICMP echo session with LAG" and explained session creation process.